platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/StartupStateUtils.java

```diff
@@ -0,0 +1,586 @@
+     * @return true if the state is suitable to be an initial state for emergency recovery, false otherwise
+     */
+    private static boolean isSuitableInitialRecoveryState(
+            @NonNull final EmergencyRecoveryManager emergencyRecoveryManager,
+            @NonNull final SavedStateInfo savedStateFile) {
+
+        if (savedStateFile.metadata().hash() == null) {
+            // This state was created with an old version of the metadata, do not consider it.
+            // Any state written with the current software version will have a non-null value for this field.
+            return false;
+        }
+
+        final Hash targetEpoch =
+                emergencyRecoveryManager.getEmergencyRecoveryFile().hash();
+        final long targetRound =
+                emergencyRecoveryManager.getEmergencyRecoveryFile().round();
+
+        final Hash stateHash = savedStateFile.metadata().hash();
+        final Hash stateEpoch = savedStateFile.metadata().epochHash();
+        final long stateRound = savedStateFile.metadata().round();
+
+        final boolean isStateSuitable = isInHashEpoch(targetEpoch, stateHash, stateEpoch) || stateRound < targetRound;
+
+        if (isStateSuitable) {
+            logger.info(
+                    STARTUP.getMarker(),
+                    """
+                            The following state meets the emergency recovery criteria:
+                                File path:     {}
+                                Hash:          {}
+                                Hash Mnemonic: {}
+                                Round:         {}""",
+                    savedStateFile.stateFile(),
+                    savedStateFile.metadata().hash(),
+                    savedStateFile.metadata().hashMnemonic(),
+                    savedStateFile.metadata().round());
+        } else {
+            logger.warn(
+                    STARTUP.getMarker(),
+                    """
+                            The following state does not meet the emergency recovery criteria:
+                                File path:     {}
+                                Hash:          {}
+                                Hash Mnemonic: {}
+                                Round:         {}""",
+                    savedStateFile.stateFile(),
+                    savedStateFile.metadata().hash(),
+                    savedStateFile.metadata().hashMnemonic(),
+                    savedStateFile.metadata().round());
+        }
+
+        return isStateSuitable;
+    }
+
+    /**
+     * Check if a provided state is in the hash epoch that is specified by the emergency recovery file.
+     *
+     * @param targetEpoch the hash epoch specified by the emergency recovery file
+     * @param stateHash   the hash of the state
+     * @param stateEpoch  the epoch hash of the state
+     * @return true if the state is in the hash epoch, false otherwise. Null states are not in the hash epoch.
+     */
+    private static boolean isInHashEpoch(
```

Comment:
I think that this is the first time I'm seeing this being called a "hash epoch". Isn't it just an "epoch", which is represented with a hash?